### DISCUSSION TITLE
[Feature] Confirmation when trying to trash picture(s)

### DISCUSSION THREAD

**Discussion by RenautMestdagh** [immich-app/immich#5864] [Feature Request]
### The feature

It would be nice to have a confirmation box when trashing pictures

### Platform

- [X] Mobile
- [ ] Server
- [X] Web

**Comment by shenlong-tanwen** [comment]
You mean while trashing the assets? We do have confirmation before permanently removing an asset.

**Comment by RenautMestdagh** [comment]
Yes, i ment when trashing assets. Shouldn't there be a confirmation for trashing as well?

**Reply by shenlong-tanwen** [reply]
> Yes, i meant when trashing assets. Shouldn't there be a confirmation for trashing as well?

Yes, A confirmation for trashing would've been better. Let's wait on others opinion on this as well.

**Reply by buha** [reply]
i agree, the trash button is too conveniently placed and accidental trashing can easily happen so confirmation is needed

**Comment by charettepa** [comment]
Yes I agree

Please add a delete confirmation dialogue when deleting photos

I understand there is a dialogue when deleting from the Trash
but there should be one when performing the initial delete

its too easy to accidentally delete
if you don't realize you did it then you could lose that photo when the set days has past in the trash

the message could be similar but state its going to trash instead of saying permanently delete
"Are you sure you want to delete? it will remain in the trash for X days unless deleted from there."

**Comment by jirip** [comment]
If I understand correctly, I disagree.
I often go through my photos and remove ones I don't want to keep, it would be very annoying to have to click "yes, I really want to delete this" for each photo.

you see popup when a photo was trashed, right? You also see it disappear from the view. If you made mistake you can go to trash and undo that.

**Comment by charettepa** [comment]
@jirip 
There is no popup that it was trashed
That's the issue

As well, while i understand you want the convenience of quick, most want the safety

This could easily be selectable just like the confirmation from trash for permanent deletion
"skip notify next time" and in the trash settings there could be a slider for notify on delete

This would solve it both ways

**Reply by jirip** [reply]
i just tried trashing a photo and i got Info popup "Moved to trash".

**Comment by charettepa** [comment]
I just re-tried 2 different ways

the popup only happens when you are in the grid select the photo then ellipsis and delete
in this scenario, you know you did it

but when in the photo and you hit the trash can icon at the top
there is no banner/popup

either way adding the "option" to get/not get the dialogue is win/win
it hurts no one
it works for everyone
if you don't want it, have it off
if you do want it, have it on

but not having the option is problematic for those that want safety (which, safety is their goal for stable release)
and if you don't you could just have the option off

**Reply by jirip** [reply]
hmm, true, the popup doesn't show up in the photo detail view, seems it is triggered though, if you have 1 photo open, delete it, go back to grid view, the popup is there...maybe it should appear even in the photo detail, regardless of any other added features.

but you are right. I like having more options too.

**Comment by glusvin** [comment]
I am experiencing this issue as well in the current Immich mobile app on Android.

The trash/delete button is located very close to the back button when viewing a photo. I have accidentally tapped the delete button several times when trying to go back, and multiple photos have ended up being deleted without any confirmation dialog.

I think this is a potentially dangerous UX issue. There should be an optional confirmation dialog such as:

“Are you sure you want to move this photo to the trash?”

Ideally, this could be configurable in Settings so users who prefer faster deletion could disable the confirmation.

Another possible improvement would be moving the delete button farther away from the back/navigation button.

This has already caused me to accidentally delete several photos, so I would strongly support adding some protection against accidental deletion.
